src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageHandlerTest.java

```diff
@@ -60,4 +68,58 @@ public void should_call_payhub_api_for_successful_payment_message() {
         assertThat(result).isNotNull();
         assertThat(result.paymentDcns).hasSize(1).contains("1234");
     }
+
+    @Test
+    public void should_call_payhub_api_to_assign_case_ref() {
+        // given
+        UpdatePaymentMessage message = new UpdatePaymentMessage(
+            "env-id-12321",
+            "Divorece",
+            "Finrem",
+            "exp-21321",
+            "cas-ref-9999"
+        );
+
+        when(s2sTokenGenerator.generate()).thenReturn("test-service");
+
+        when(payHubClient.updateCaseReference(
+            eq("test-service"), eq("exp-21321"), any(CaseReferenceRequest.class))
+        ).thenReturn(ResponseEntity.status(200).build());
+
+        // when
+        messageHandler.updatePaymentCaseReference(message);
+
+        // then
+        ArgumentCaptor<CaseReferenceRequest> requestCaptor = ArgumentCaptor.forClass(CaseReferenceRequest.class);
+
+        verify(payHubClient).updateCaseReference(any(), any(), requestCaptor.capture());
+        CaseReferenceRequest req = requestCaptor.getValue();
+        CaseReferenceRequest expectedRequest = new CaseReferenceRequest("cas-ref-9999");
+        assertThat(req).isEqualToComparingFieldByFieldRecursively(expectedRequest);
+    }
+
+
+    @Test
+    public void should_throw_exception_if_payhub_api_assign_case_ref_fails() {
+        // given
+        UpdatePaymentMessage message = new UpdatePaymentMessage(
+            "env-id-12321",
+            "Divorece",
+            "Finrem",
+            "exp-21321",
+            "cas-ref-9999"
+        );
+
+        when(s2sTokenGenerator.generate()).thenReturn("test-service");
+
+        when(payHubClient.updateCaseReference(
+            eq("test-service"), eq("exp-21321"), any(CaseReferenceRequest.class))
+        ).thenThrow(new PayHubClientException(new HttpClientErrorException(HttpStatus.BAD_REQUEST, "BAD_REQUEST")));
+
+        // when
+        assertThatThrownBy(
+            () -> messageHandler.updatePaymentCaseReference(message))
+            .isInstanceOf(PayHubClientException.class);
```

Comment:
This test is not needed in my opinion

Reply:
was about to say "on the other hand" similarly. yes, there's no edge case at the moment to verify. perhaps somewhere higher in the food chain when we need to verify exception handling